pkg/gameservers/controller.go

```diff
@@ -124,20 +128,25 @@ func NewController(
 	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
 	c.recorder = eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "gameserver-controller"})
 
-	c.workerqueue = workerqueue.NewWorkerQueue(c.syncGameServer, c.logger, stable.GroupName+".GameServerController")
+	c.workerqueue = workerqueue.NewWorkerQueueWithRateLimiter(c.syncGameServer, c.logger, stable.GroupName+".GameServerController",
```

Comment:
So they all have the same settings, and the same sync - I'm dreadfully curious - how does this improve things in the `GameServer` controller?

Is it just that creation and deletion get's pushed to the front of it's own queue, while there rest can happen at a separate priority? (or at the same time?)

Reply:
that's exactly correct. GSS can then add/delete lots of GS very quickly without clogging the processing of GS already in-flight.